### ISSUE TITLE
Incsositent behavior with data.type and data.types when dynamically loading data

### ISSUE THREAD

**Issue by danelkhen** [masayuki0812/c3#1013]
Hi,
When creating a bar chart (data.type='bar'), with some line graph (data.types={ddd:"line"}) - initially all works ok, but when loading the same data using chart.load() - all bars turn into lines. Perhaps it's illegal to use data.type and data.types? I'm using the fact that data.type means that default type, and data.types allows you to customize specific names. When specifying all target names in data.types all works ok. I'm just wondering if this should be a bit more consistent - to have the same behavior when using new c3.generate() and chart.load().

```
var chart = c3.generate({
    axis: { x: { type: "categorized", } },
    data: {
        type: "bar",
        types: {"ddd": "line"},
        columns: [
            ["aaa", "a", "b", "c"],
            ["bbb", 10,20,30],
            ["ccc", 40,50,60],
            ["ddd", 70,80,90],
        ],
        x: "aaa",
        groups: [["aaa", "bbb", "ccc"]],
    },
});

window.setTimeout(function(){
chart.load({
    type: "bar",
    types: {"ddd": "line"}
    columns: [
            ["aaa", "a", "b", "c"],
            ["bbb", 30,20,30],
            ["ccc", 80,50,60],
            ["ddd", 20,80,90],
    ],
    x: "aaa",
    groups: [["aaa", "bbb", "ccc"]],
});
}, 2000);
```

**Comment by masayuki0812** [comment]
Hi, Thanks for reporting. You're right and I think this has been fixed now. I'll release this fix in the next version 0.4.10 shortly. Thanks!

**Comment by masayuki0812** [comment]
`0.4.10` has been released, so let me close at the moment. Thanks.
